Sources/VectorExtor/CGPath+Extensions.swift

```diff
@@ -3,275 +3,152 @@ import CoreGraphics
+				curve = .addCurveTo(.init(startPoint: prevEnd, control1: control1, control2: control2, endPoint: point))
+				setClose(prevEnd)
 			case .closeSubpath:
-				store.append(.close)
+				curve = .close(.init(startPoint: prevEnd, endPoint: closeEnd ?? .zero))
+				closeEnd = nil
 			@unknown default:
 				print("Unknown path element type: \(element.type) \(element.type.rawValue)")
+				return
 			}
-		}
-		return store
-	}
 
-	var sections: [PathSection] {
-		var store = [PathSection]()
-		var previous: PathSection?
-		for element in elements {
-			let newSection = PathSection(element: element)
-			newSection.previous = previous
-			previous?.next = newSection
-			previous = newSection
-			store.append(newSection)
+			let new = Segment(curve: curve)
+			accumulator.append(new)
 		}
-		return store
+		return accumulator
 	}
 
-	var length: CGFloat { sections.length }
+	var length: CGFloat { segments.length }
 
-	var svgString: String {
-		toSVGString()
-	}
+	var svgString: String { toSVGString() }
 
 	private func toSVGString() -> String {
-		let components = elements
+		let components = segments
 
 		return components.reduce("") {
-			switch $1 {
-			case .moveTo(point: let point):
-				return $0 + "M\(point.x),\(point.y) "
-			case .addLineTo(point: let point):
-				return $0 + "L\(point.x),\(point.y) "
-			case .addQuadCurveTo(point: let point, controlPoint: let control):
-				return $0 + "Q\(control.x),\(control.y) \(point.x),\(point.y) "
-			case .addCurveTo(point: let point, controlPoint1: let control1, controlPoint2: let control2):
-				return $0 + "C\(control1.x),\(control1.y) \(control2.x),\(control2.y) \(point.x),\(point.y) "
+			switch $1.curve {
+			case .moveTo(let move):
+				return $0 + "M\(move.endPoint.x),\(move.endPoint.y) "
+			case .addLineTo(let line):
+				return $0 + "L\(line.endPoint.x),\(line.endPoint.y) "
+			case .addQuadCurveTo(let quad):
+				return $0 + "Q\(quad.controlPoint.x),\(quad.controlPoint.y) \(quad.endPoint.x),\(quad.endPoint.y) "
+			case .addCurveTo(let cubic):
+				return $0 + "C\(cubic.control1.x),\(cubic.control1.y) \(cubic.control2.x),\(cubic.control2.y) \(cubic.endPoint.x),\(cubic.endPoint.y) "
 			case .close:
 				return $0 + "z "
 			}
 		}
 	}
 
 	func pointAlongPath(atPercent percent: CGFloat, precalculatedLength: CGFloat? = nil) -> CGPoint? {
-		let subpaths = sections
-		let percent = percent.clamped()
-		let length = precalculatedLength ?? self.length
-		let desiredLength = length * percent
-
-		var currentLength: CGFloat = 0
-
-		for subpath in subpaths {
-			let thisSubLength = subpath.length
-			currentLength += thisSubLength
-			if currentLength >= desiredLength {
-				let remaining = desiredLength - (currentLength - thisSubLength)
-				let straightPercentage = thisSubLength != 0 ? remaining / thisSubLength : 0
-				return subpath.pointAlongCurve(atPercent: straightPercentage)
-			}
-		}
+//		let subpaths = sections
```

Comment:
delete?

Reply:
I think you must have been looking at an old commit? This and one other comment are both tagged "Outdated" by GitHub, and I haven't pushed any changes since replying yet. Anyways, this is gone now :)